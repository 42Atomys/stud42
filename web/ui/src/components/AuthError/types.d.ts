type ErrorType =
  | 'default'
  | 'configuration'
  | 'accessdenied'
  | 'verification'
  | 'oauthcreateaccount'
  | 'oauthaccountnotlinked'
  | 'oauthcallback'
  | 'callback'
  | 'sessionrequired';

type ErrorView = {
  name: string;
  message: string;
};
