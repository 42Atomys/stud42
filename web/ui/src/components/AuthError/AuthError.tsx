export const Error = ({
  error = 'default',
}: {
  error: ErrorType | 'null' | 'undefined';
}) => {
  const errors: Record<ErrorType, ErrorView> = {
    default: {
      name: 'An error occured',
      message: 'An error occured, please try again later',
    },
    configuration: {
      name: 'Configuration error',
      message: 'The configuration of the provider is not valid',
    },
    accessdenied: {
      name: 'Access denied',
      message: "You don't have access to this page",
    },
    verification: {
      name: 'Verification error',
      message: 'The verification of the provider is not valid',
    },
    oauthcreateaccount: {
      name: 'OAuth create account error',
      message: 'An error occured while creating your account',
    },
    oauthaccountnotlinked: {
      name: 'OAuth account not linked',
      message:
        'Your account is not linked to your email. Contact support on Discord',
    },
    oauthcallback: {
      name: 'OAuth callback error',
      message: 'An error occured while linking your account',
    },
    callback: {
      name: 'Callback error',
      message: 'An error occured while processing the callback',
    },
    sessionrequired: {
      name: 'Access required',
      message: 'You need to be logged in to access this page',
    },
  };

  if (error === 'null' || error === 'undefined') error = 'default';

  const { name, message } = errors[error.toLowerCase() as ErrorType];
  return (
    <div className="bg-red-400 px-6 py-4 text-white rounded mb-10">
      <h1 className="text-lg mb-2">{name}</h1>
      <p className="text-sm text-red-800 dark:text-red-100 mb-2">{message}</p>
    </div>
  );
};

export default Error;
