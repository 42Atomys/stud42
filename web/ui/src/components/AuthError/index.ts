export { AuthError, AuthError as default } from './AuthError';
