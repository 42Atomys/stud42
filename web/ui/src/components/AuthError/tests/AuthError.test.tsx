import { render } from '@testing-library/react';
import { AuthError } from '@components/AuthError';

it('snapshot: renders Tooltip unchanged', () => {
  const { container } = render(<AuthError error="default" />);
  expect(container).toMatchSnapshot();
});

it('render correct data when error is an ErrorType', async () => {
  const container = render(<AuthError error="callback" />);

  expect(container.getByText('Callback error')).toBeInTheDocument();
  expect(
    container.getByText('An error occured while processing the callback')
  ).toBeInTheDocument();
});

it('render correct data when error is not defined', async () => {
  const container = render(<AuthError error={null} />);

  expect(container.getByText('An error occured')).toBeInTheDocument();
  expect(
    container.getByText('An error occured, please try again later')
  ).toBeInTheDocument();
});

it('render correct data when error is not included', async () => {
  // @ts-ignore
  const container = render(<AuthError error="testvaluedontmatch" />);

  expect(container.getByText('An error occured')).toBeInTheDocument();
  expect(
    container.getByText('An error occured, please try again later')
  ).toBeInTheDocument();
});
