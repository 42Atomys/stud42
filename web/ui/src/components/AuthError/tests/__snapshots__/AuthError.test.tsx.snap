// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot: renders Tooltip unchanged 1`] = `
<div>
  <div
    class="bg-red-400 px-6 py-4 text-white rounded mb-10"
  >
    <h1
      class="text-lg mb-2"
    >
      An error occured
    </h1>
    <p
      class="text-sm text-red-800 dark:text-red-100 mb-2"
    >
      An error occured, please try again later
    </p>
  </div>
</div>
`;
